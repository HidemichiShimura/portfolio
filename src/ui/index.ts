// Atoms
export { default as PageHeading } from "./atoms/PageHeading/PageHeading";
export { default as Avatar } from "./atoms/Avatar/Avatar";
export { default as Typography } from "./atoms/Typography/Typography";
export { default as SectionHeading } from "./atoms/SectionHeading/SectionHeading";
export { default as Icon } from "./atoms/Icon/Icon";
export { default as RadialLinearContainer } from "./atoms/RadialLinearContainer/RadialLinearContainer";
export { default as Grid } from "./atoms/Grid/Grid";
export { default as GradientCard } from "./atoms/GradientCard/GradientCard";
export { default as Card } from "./atoms/Card/Card";

// Molecules
export { default as HeroHeadings } from "./molecules/HeroHeadings/HeroHeadings";
