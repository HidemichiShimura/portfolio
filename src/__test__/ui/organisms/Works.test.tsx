import { Works } from "@/ui";
import { screen, render } from "@testing-library/react";

const TITLE_ATTR = "Works";
const SECTION_TITLE = "WORKS";

describe("Organisms: <Works />", () => {
  test("renders", () => {
    render(<Works />);

    expect(screen.getByTitle(TITLE_ATTR)).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    render(<Works />);

    expect(screen.getByTitle(TITLE_ATTR)).toMatchSnapshot();
  });

  test("contains a section title", () => {
    render(<Works />);

    expect(screen.queryByRole("heading")).toBeInTheDocument();
  });

  test("displays the correct section title", () => {
    render(<Works />);

    expect(screen.queryByRole("heading")?.textContent).toBe(SECTION_TITLE);
  });
});
