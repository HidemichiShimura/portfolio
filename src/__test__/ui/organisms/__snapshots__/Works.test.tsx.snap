// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms: <Works /> matches snapshot 1`] = `
<section
  class="flex min-h-screen flex-col justify-center gap-8"
  title="Works"
>
  <h2
    class="bg-gradient-to-b from-black to-neutral-100 bg-clip-text font-sans text-4xl text-transparent dark:from-neutral-100 dark:to-black"
  >
    WORKS
  </h2>
  <div
    class="flex flex-col justify-center gap-4"
  >
    <div
      class="overflow-hidden rounded-lg"
    >
      <div
        class="w-full h-96 bg-cover bg-center bg-no-repeat bg-pf"
      >
        <div
          class="flex h-full w-full flex-col items-center justify-center gap-4 bg-slate-900 opacity-50"
        >
          <p
            class="text-white"
          >
            Portfolio
          </p>
          <ul
            class="flex gap-2"
          >
            <li
              class="w-7"
            >
              <a
                href="https://github.com/HidemichiShimura/portfolio"
                target="_blank"
              >
                <img
                  class="invert"
                  src="/img.jpg"
                />
              </a>
            </li>
            <li
              class="w-7"
            >
              <a
                href="https://portfolio-hidemichishimura.vercel.app/"
                target="_blank"
              >
                <img
                  class="invert"
                  src="/img.jpg"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="overflow-hidden rounded-lg"
    >
      <div
        class="w-full h-96 bg-cover bg-center bg-no-repeat bg-wt"
      >
        <div
          class="flex h-full w-full flex-col items-center justify-center gap-4 bg-slate-900 opacity-50"
        >
          <p
            class="text-white"
          >
            Weight Tracker
          </p>
          <ul
            class="flex gap-2"
          >
            <li
              class="w-7"
            >
              <a
                href="https://github.com/HidemichiShimura/weight-tracker"
                target="_blank"
              >
                <img
                  class="invert"
                  src="/img.jpg"
                />
              </a>
            </li>
            <li
              class="w-7"
            >
              <a
                href="https://weight-tracker-front.vercel.app/"
                target="_blank"
              >
                <img
                  class="invert"
                  src="/img.jpg"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="overflow-hidden rounded-lg"
    >
      <div
        class="w-full h-96 bg-cover bg-center bg-no-repeat bg-cdc"
      >
        <div
          class="flex h-full w-full flex-col items-center justify-center gap-4 bg-slate-900 opacity-50"
        >
          <p
            class="text-white"
          >
            Cafe Drink Customizer
          </p>
          <ul
            class="flex gap-2"
          >
            <li
              class="w-7"
            >
              <a
                href="https://github.com/HidemichiShimura/cafe-drink-customizer"
                target="_blank"
              >
                <img
                  class="invert"
                  src="/img.jpg"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="overflow-hidden rounded-lg"
    >
      <div
        class="w-full h-96 bg-cover bg-center bg-no-repeat bg-scb"
      >
        <div
          class="flex h-full w-full flex-col items-center justify-center gap-4 bg-slate-900 opacity-50"
        >
          <p
            class="text-white"
          >
            Slack Cocktail Bot
          </p>
          <ul
            class="flex gap-2"
          >
            <li
              class="w-7"
            >
              <a
                href="https://github.com/HidemichiShimura/slack-cocktail-bot"
                target="_blank"
              >
                <img
                  class="invert"
                  src="/img.jpg"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</section>
`;
